improver/developer_tools/metadata_interpreter.py

```diff
@@ -518,7 +518,21 @@ def run(self, cube: Cube) -> None:
         except ValueError as cause:
             self.errors.append(str(cause))
 
-        # 6) Raise collated errors if present
+        # 6) Check multiple realizations only exist for ensemble models
+        if self.field_type == self.DIAG:
+            try:
+                realization_coord = cube.coord("realization")
+            except CoordinateNotFoundError:
+                pass
+            else:
+                model_id = cube.attributes.get(self.model_id_attr, "ens")
```

Comment:
I think this is okay. If model_id_attr is not set that will be flagged by a test above, and once fixed and the test is re-run on the file, if it's inappropriate this will then trigger.